pandas/core/apply.py

```diff
@@ -19,20 +24,31 @@ def frame_apply(obj, func, axis=0, broadcast=False,
         klass = FrameColumnApply
 
     return klass(obj, func, broadcast=broadcast,
-                 raw=raw, reduce=reduce, args=args, kwds=kwds)
+                 raw=raw, reduce=reduce, result_type=result_type,
+                 ignore_failures=ignore_failures,
+                 args=args, kwds=kwds)
 
 
 class FrameApply(object):
 
-    def __init__(self, obj, func, broadcast, raw, reduce, args, kwds):
+    def __init__(self, obj, func, broadcast, raw, reduce, result_type,
+                 ignore_failures, args, kwds):
         self.obj = obj
-        self.broadcast = broadcast
         self.raw = raw
         self.reduce = reduce
-        self.args = args
+        self.ignore_failures = ignore_failures
+        self.args = args or ()
+        self.kwds = kwds or {}
 
-        self.ignore_failures = kwds.pop('ignore_failures', False)
-        self.kwds = kwds
+        if broadcast is not None:
+            warnings.warn("The broadcast argument is deprecated and will "
+                          "be removed in a future version. You can specify "
+                          "result_type='broadcast' broadcast a scalar result",
```

Comment:
Missing a "to" before "broadcast a scalar result"?

Reply:
updated